src/main/java/seedu/clinic/logic/commands/AddCommand.java

```diff
@@ -3,65 +3,97 @@
+            + PREFIX_SUPPLIER_NAME + "NAME "
             + PREFIX_PHONE + "PHONE "
             + PREFIX_EMAIL + "EMAIL "
-            + PREFIX_ADDRESS + "ADDRESS "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + PREFIX_REMARK + "REMARK "
             + "Example: " + COMMAND_WORD + " "
-            + PREFIX_NAME + "John Doe "
+            + PREFIX_SUPPLIER_NAME + "John Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
-            + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
-            + PREFIX_TAG + "friends "
-            + PREFIX_TAG + "owesMoney";
+            + PREFIX_REMARK + "Largest contractor \n"
+            + "Warehouse Parameters: "
+            + PREFIX_WAREHOUSE_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_ADDRESS + "Address "
+            + PREFIX_REMARK + "REMARK "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_WAREHOUSE_NAME + "NUS South "
+            + PREFIX_PHONE + "91234562 "
+            + PREFIX_ADDRESS + "21 Lower Kent Ridge Rd, Singapore 119077 "
+            + PREFIX_REMARK + "Mainly deal with schools ";
 
-    public static final String MESSAGE_SUCCESS = "New supplier added: %1$s";
-    public static final String MESSAGE_DUPLICATE_SUPPLIER = "This supplier already exists in the CLI-nic app";
+    public static final String MESSAGE_SUPPLIER_SUCCESS = "New supplier added: %1$s";
+    public static final String MESSAGE_WAREHOUSE_SUCCESS = "New warehouse added: %1$s";
+    public static final String MESSAGE_DUPLICATE_SUPPLIER = "This supplier already exists in CLI-nic";
+    public static final String MESSAGE_DUPLICATE_WAREHOUSE = "This warehouse already exists in CLI-nic";
 
-    private final Supplier toAdd;
+    private final Supplier supplierToAdd;
+    private final Warehouse warehouseToAdd;
 
     /**
      * Creates an AddCommand to add the specified {@code Supplier}
      */
     public AddCommand(Supplier supplier) {
         requireNonNull(supplier);
-        toAdd = supplier;
+        this.supplierToAdd = supplier;
+        this.warehouseToAdd = null;
+    }
+
+    /**
+     * Creates an AddCommand to add the specified {@code Warehouse}
+     */
+    public AddCommand(Warehouse warehouse) {
+        requireNonNull(warehouse);
+        this.supplierToAdd = null;
+        this.warehouseToAdd = warehouse;
+
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (model.hasSupplier(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_SUPPLIER);
+        if (supplierToAdd != null) {
+            if (model.hasSupplier(supplierToAdd)) {
+                throw new CommandException(MESSAGE_DUPLICATE_SUPPLIER);
+            }
+            model.addSupplier(supplierToAdd);
+            return new CommandResult(String.format(MESSAGE_SUPPLIER_SUCCESS, supplierToAdd));
+        } else {
```

Comment:
perhaps its better code quality to explicitly specify the else with a separate if statement.

Reply:
Thanks for pointing it out, I do agree with you and have specified in the next commit!